static/js/subs.js

```diff
@@ -244,14 +244,21 @@ exports.update_settings_for_unsubscribed = function (sub) {
     var active_stream = exports.active_stream();
     if (active_stream !== undefined && active_stream.id === sub.stream_id) {
         stream_edit.rerender_subscribers_list(sub);
-    }
 
-    // If user unsubscribed from private stream then user can not subscribe to
-    // stream without invitation. So hide subscribe button.
-    stream_data.update_calculated_fields(sub);
-    if (!sub.should_display_subscription_button) {
-        settings_button.hide();
+        // Remove stream from subscribed list.
+        if ($("#subscriptions_table .search-container .tab-switcher .first").hasClass("selected")) {
+            var sub_row = $('#subscriptions_table .stream-row[data-stream-id=' + sub.stream_id + ']');
+            sub_row.addClass("notdisplayed");
+        }
```

Comment:
This condition check if currently `Subscribed` tab is active, not `All stream` tab. Cause we don't want to remove stream from `All stream` tab.